src/OpenApi/src/Extensions/OpenApiEndpointRouteBuilderExtensions.cs

```diff
@@ -49,8 +49,16 @@ public static IEndpointConventionBuilder MapOpenApi(this IEndpointRouteBuilder e
                     using var writer = Utf8BufferTextWriter.Get(output);
                     try
                     {
-                        document.Serialize(new OpenApiJsonWriter(writer), documentOptions.OpenApiVersion);
-                        context.Response.ContentType = "application/json;charset=utf-8";
+                        if (UseYaml(pattern))
+                        {
+                            document.Serialize(new OpenApiYamlWriter(writer), documentOptions.OpenApiVersion);
+                            context.Response.ContentType = "application/yaml;charset=utf-8";
```

Comment:
[RFC 9512](https://www.rfc-editor.org/rfc/rfc9512.html) declares `application/yaml` as the official content-type for YAML-based types. One thing I observed is that this MIME type is fairly new and most browsers still don't know how to render it inline and will fallback to downloading the YAML file to disk.

To get around this, we could use the `text/plain+yaml` content type to force inline rendering while still being in compliance with the MIME type spec for YAML.

Side note: This [RFC](https://datatracker.ietf.org/doc/draft-ietf-httpapi-rest-api-mediatypes/) is in draft which defines an official `application/openapi` content type that we might consider using in the future.

Reply:
Based on my limited knowledge of MIME types, `application/yaml` seems more correct since it is primarily for programmatic consumption, but I can definitely imagine people asking for `text+yaml` as a convenience.  I see one of the RFC authors was from Mozilla, so hopefully built-in support isn't too far off?